GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -373,6 +375,75 @@ void ManageWorktreeSupport()
             }
         }
 
+        private readonly Regex _aheadBehindRegEx = new Regex(@"^\[(ahead (?<ahead>\d+))?(, )?(behind (?<behind>\d+))?\] (?<branch>.*)$", RegexOptions.Compiled | RegexOptions.Multiline);
+
+        private readonly TranslationString _aheadCommitsToPush = new TranslationString("{0} new commit(s) will be pushed");
+        private readonly TranslationString _behindCommitsTointegrateOrForcePush = new TranslationString("{0} commit(s) should be integrated (or will be lost if force pushed)");
+
+        private const string aheadBehindGitCommand = "for-each-ref --format=\"%(push:track) %(refname:short)\" refs/heads/";
+        private void DisplayAheadBehindInformation()
+        {
+            if (!GitVersion.Current.SupportAheadBehindData)
+            {
+                return;
+            }
+
+            toolStripButtonPush.DisplayStyle = ToolStripItemDisplayStyle.Image;
+            toolStripButtonPush.Image = Images.Push;
+            toolStripButtonPush.ToolTipText = _push.Text;
+
+            var currentBranch = UICommands.GitModule.GetSelectedBranch();
+            if (currentBranch == "(no branch)")
+            {
+                return;
+            }
+
+            var result = UICommands.GitModule.RunGitCmd(aheadBehindGitCommand + currentBranch);
```

Comment:
Use gitargumentbuilder.

Reply:
```suggestion
    var cmd= new GitArgumentBuilder("for-each-ref")
    {
     "--format=\"%(push:track) %(refname:short)\"",
     "refs/heads/",
     currentBranch
    };
            var result = UICommands.GitModule.RunGitCmd(cmd);
```

Something like that.

Reply:
I updated the wiki.

https://github.com/gitextensions/gitextensions/wiki#coding-guide